src/components/core/GeoChart.tsx

```diff
@@ -70,39 +103,136 @@ export class GeoChartInner extends BaseVisualization<IGeoChartInnerProps, {}> {
         );
     }
 
+    public onLegendItemClick(item: any) {
+        this.setState({
+            legendIndexEnabled: set<boolean[]>(
+                [...this.state.legendIndexEnabled],
+                item.legendIndex,
+                !this.state.legendIndexEnabled[item.legendIndex],
+            ),
+            legendNamesEnabled: this.getSelectedSegmentItem(
+                item.name,
+                !this.state.legendIndexEnabled[item.legendIndex],
+            ),
+        });
+    }
+
+    private getSelectedSegmentItem(name: string, visible: boolean): string[] {
+        const selectedSegmentItem: string[] = [...this.state.legendNamesEnabled];
+        if (visible) {
+            selectedSegmentItem.push(name);
+        } else {
+            const index = selectedSegmentItem.indexOf(name);
+            selectedSegmentItem.splice(index, 1);
+        }
+        return selectedSegmentItem;
+    }
+
+    private getSegmentBySeries(data: Execution.IResultHeaderItem[]): string[] {
+        const segmentItemNames: string[] = data.map(
+            (item: Execution.IResultHeaderItem): string => {
+                const name: string = getHeaderItemName(item);
+                return name || EMPTY_SEGMENT_ITEM;
+            },
+        );
+        return uniq(segmentItemNames);
+    }
+
+    private transformationData() {
```

Comment:
We can rename it to `getGeoChartOptions`

Reply:
fixed